comp/api/api/apiimpl/internal/agent/agent.go

```diff
@@ -87,11 +89,16 @@ func SetupHandlers(
 	collector optional.Option[collector.Component],
 	eventPlatformReceiver eventplatformreceiver.Component,
 	ac autodiscovery.Component,
+	providers []api.EndpointProvider,
 ) *mux.Router {
 
+	// TODO: move these to a component that is registerable
+	for _, p := range providers {
+		r.Handle(p.Route, p.Handler).Methods(p.Methods...)
+	}
 	r.HandleFunc("/version", common.GetVersion).Methods("GET")
 	r.HandleFunc("/hostname", getHostname).Methods("GET")
-	r.HandleFunc("/flare", func(w http.ResponseWriter, r *http.Request) { makeFlare(w, r, flareComp) }).Methods("POST")
+	//r.HandleFunc("/flare", func(w http.ResponseWriter, r *http.Request) { makeFlare(w, r, flareComp) }).Methods("POST")
```

Comment:
🔥 🔥

Reply:
removed in https://github.com/DataDog/datadog-agent/pull/24045/commits/dd24bb32483d68bc40fbebc5a077395d8fa2a4a4